### PR TITLE
nvme: Add Media Unit Status log page(LID: 0x10)

### PR DESCRIPTION
**Pull request by Arunpandian15**
Signed-off-by: Arunpandian J <apj.arun@samsung.com>

### CODE REVIEW

File: src/nvme/types.h [hunk 1]
```diff
@@ -6024,4 +6026,30 @@ enum nvme_zns_report_options {
...
+};
+/**
+ * struct nvme_media_unit_stat_log 
+ */
+
+struct nvme_media_unit_stat_log {
+	__le16	nmu;
+	__le16	cchans;
+	__le16	sel_config;
+	__u8	rsvd1[10];
```
Comment (sc108-lee) [inline src/nvme/types.h:6051]: rsvd6[10];

File: src/nvme/types.h [hunk 2]
```diff
@@ -5322,6 +5323,7 @@ enum nvme_cmd_get_log_lid {
 	NVME_LOG_LID_PERSISTENT_EVENT				= 0x0d,
 	NVME_LOG_LID_LBA_STATUS					= 0x0e,
 	NVME_LOG_LID_ENDURANCE_GRP_EVT				= 0x0f,
+	NVME_LOG_LID_MEDIA_UNIT_STATUS              = 0x10,
```
[the remark this answers is not in the document]
Comment (Arunpandian15) [reply src/nvme/types.h]: Addressed
Thanks @sc108-lee

File: src/nvme/ioctl.c [hunk 3]
```diff
@@ -858,6 +858,13 @@ int nvme_get_log_zns_changed_zones(int fd, __u32 nsid, bool rae,
 			    NULL);
 }
 
+int nvme_media_unit_stat_log(int fd,  __u16 domid, struct nvme_media_unit_stat_log *mus)
+{
+	return nvme_get_log(fd, NVME_LOG_LID_MEDIA_UNIT_STATUS, NVME_NSID_NONE, 0,
+		NVME_LOG_LSP_NONE, domid, 0, NVME_UUID_NONE, NVME_CSI_NVM,
+		sizeof(*mus), mus);
```
[the remark this answers is not in the document]
Comment (Arunpandian15) [reply src/nvme/ioctl.c]: Addressed
Thanks @sc108-lee

File: src/nvme/types.h (hunk 1 above)
Comment (Arunpandian15) [reply src/nvme/types.h:6051]: Addressed
Thanks @sc108-lee

File: src/nvme/ioctl.h [hunk 4]
```diff
@@ -1133,6 +1133,15 @@ int nvme_get_log_zns_changed_zones(int fd, __u32 nsid, bool rae,
 int nvme_get_log_persistent_event(int fd, enum nvme_pevent_log_action action,
 				  __u32 size, void *pevent_log);
 
+/**
+ * nvme_media_unit_stat_log() -
+ * &fd: File descriptor of nvme device
+ * &domid: Domain Identifier
+ * &mus: User address to store the media unit status log
+ */
+
```
Comment (igaw) [inline src/nvme/ioctl.h:1142]: - no new line between the doc and the function.
- please align the documentation like the other function have.

File: src/nvme/ioctl.h [hunk 5]
```diff
@@ -1133,6 +1133,15 @@ int nvme_get_log_zns_changed_zones(int fd, __u32 nsid, bool rae,
...
+int nvme_media_unit_stat_log(int fd, __u16 domid, struct nvme_media_unit_stat_log *mus);
```
Comment (igaw) [inline src/nvme/ioctl.h:1143]: Overlong line

File: src/nvme/ioctl.c [hunk 6]
```diff
@@ -858,6 +858,13 @@ int nvme_get_log_zns_changed_zones(int fd, __u32 nsid, bool rae,
 			    NULL);
 }
 
+int nvme_media_unit_stat_log(int fd,  __u16 domid, struct nvme_media_unit_stat_log *mus)
+{
+	return nvme_get_log(fd, NVME_LOG_LID_MEDIA_UNIT_STATUS, NVME_NSID_NONE, 0,
+		NVME_LOG_LSP_NONE, domid, 0, NVME_UUID_NONE, NVME_CSI_NVM, false,
```
Comment (igaw) [inline src/nvme/ioctl.c:864]: - whitespace at the end of line
- align the start of the line to the (
-

File: src/nvme/types.h [hunk 7]
```diff
@@ -6024,4 +6026,30 @@ enum nvme_zns_report_options {
 	NVME_ZNS_ZRAS_REPORT_OFFLINE		= 0x7,
 };
 
+/**
+ * struct nvme_media_unit_stat_desc
```
Comment (igaw) [inline src/nvme/types.h:6030]: - whitespace at the end of the line
- please add some documentation for each member.

File: src/nvme/types.h [hunk 8]
```diff
@@ -6024,4 +6026,30 @@ enum nvme_zns_report_options {
...
+	__le16	domainid;
+	__le16	endgid;
+	__le16	nvmsetid;
+	__le16	cap_adj_fctr;
+	__u8	avl_spare;
+	__u8	percent_used;
+	__u8	mucs;
+	__u8	cio;
+};
+/**
```
Comment (igaw) [inline src/nvme/types.h:6043]: add a new line here

File: src/nvme/types.h [hunk 9]
```diff
@@ -6024,4 +6026,30 @@ enum nvme_zns_report_options {
...
+ * struct nvme_media_unit_stat_log 
+ */
+
```
Comment (igaw) [inline src/nvme/types.h:6046]: remove a new line here

File: src/nvme/types.h [hunk 10]
```diff
@@ -6024,4 +6026,30 @@ enum nvme_zns_report_options {
...
+	__le16	nvmsetid;
+	__le16	cap_adj_fctr;
+	__u8	avl_spare;
+	__u8	percent_used;
+	__u8	mucs;
+	__u8	cio;
+};
+/**
+ * struct nvme_media_unit_stat_log 
+ */
```
Comment (igaw) [inline src/nvme/types.h:6045]: please document each member.

**Review by igaw — CHANGES_REQUESTED** [review]
- Mostly formatting nitpicking.
- Please add some documentation. I know we don't have documented everything but let's add some whenever we add new stuff.

Thanks for your contribution!

File: src/nvme/ioctl.h (hunk 4 above)
Comment (Arunpandian15) [reply src/nvme/ioctl.h:1142]: Addressed
Thanks @igaw

File: src/nvme/ioctl.h (hunk 5 above)
Comment (Arunpandian15) [reply src/nvme/ioctl.h:1143]: Addressed
Thanks @igaw

File: src/nvme/ioctl.c (hunk 6 above)
Comment (Arunpandian15) [reply src/nvme/ioctl.c:864]: Addressed
Thanks @igaw

File: src/nvme/types.h (hunk 7 above)
Comment (Arunpandian15) [reply src/nvme/types.h:6030]: Addressed
Thanks @igaw

File: src/nvme/types.h (hunk 9 above)
Comment (Arunpandian15) [reply src/nvme/types.h:6046]: Addressed
Thanks @igaw

File: src/nvme/types.h (hunk 10 above)
Comment (Arunpandian15) [reply src/nvme/types.h:6045]: Addressed
Thanks @igaw

File: src/nvme/types.h (hunk 8 above)
Comment (Arunpandian15) [reply src/nvme/types.h:6043]: Addressed
Thanks @igaw

**Comment by igaw** [comment]
Looks good. Again I'd like to get @hreinecke's #150 sorted out first.

**Comment by Arunpandian15** [comment]
@igaw 
I have rebased my PR with latest repo, Please check. Thanks!

**Pull request merged**